Tab hook docs (#1888)
adobe/react-spectrum: packages/@react-aria/tabs/docs/useTabList.mdx

Each tab can be clicked, tapped, or navigated to via arrow keys. Depending on the `keyboardActivation` prop, the tab can be selected by receiving keyboard focus, or it can be selected with the <kbd>Enter</kbd> key.

`useTabList` returns props to spread onto the tab list container:

<TypeContext.Provider value={docs.links}>
  <InterfaceType properties={docs.links[docs.exports.useTabList.return.id].properties} />
</TypeContext.Provider>

`useTab` returns props to be spread onto each individual tab:

<TypeContext.Provider value={docs.links}>
  <InterfaceType properties={docs.links[docs.exports.useTab.return.id].properties} />
</TypeContext.Provider>

`useTabPanel` returns props to spread onto the container for the tab content:

<TypeContext.Provider value={docs.links}>
  <InterfaceType properties={docs.links[docs.exports.useTabPanel.return.id].properties} />
</TypeContext.Provider>

State is managed by the <TypeLink links={statelyDocs.links} type={statelyDocs.exports.useTabListState} />
hook in `@react-stately/tabs`. The state object should be passed as an option to `useTabList`, `useTab`,
and `useTabPanel`.

## Example

> This example displays a basic list of tabs. The currently selected tab get a `tabIndex` of 0 while the rest are set to -1 ensuring that the tablist is only a single tab stop. The selected tab gets a different style so it's obvious which one is currently selected. `useTab` and `useTabPanel` handle associating the tabs and tab panels for assistive technology. The currently selected tab panel is rendered below the list of tabs. The `key` prop on the `TabPanel` element is important to ensure that DOM state (e.g. text field contents) is not shared between unrelated tabs.

Reviewer: ```suggestion
This example displays a basic list of tabs. The currently selected tab receives a `tabIndex` of 0 while the rest are set to -1 ensuring that the whole tablist is a single tab stop. The selected tab has a different style so it's obvious which one is currently selected. `useTab` and `useTabPanel` handle associating the tabs and tab panels for assistive technology. The currently selected tab panel is rendered below the list of tabs. The `key` prop on the `TabPanel` element is important to ensure that DOM state (e.g. text field contents) is not shared between unrelated tabs.
```
